File: src/transforms.rs
use crate::regex_constants;
use regex::Regex;
use unidecode::unidecode;

pub fn to_lower(text: &str) -> String {
    text.to_lowercase()
}

pub fn remove_accents(text: &str) -> String {
    unidecode(text)
}

pub fn remove_punctuation(text: &str) -> String {
    regex_constants::PUNCTUATION_REGEX
        .replace_all(text, " ")
        .to_string()
}

pub fn remove_blank_spaces(text: &str) -> String {
    text.split_whitespace().collect::<Vec<&str>>().join(" ")
}

pub fn remove_html(text: &str) -> String {
    regex_constants::HTML_REGEX
        .replace_all(text, " ")
        .to_string()
}

pub fn replace_escape_characters(
    text: &str,
    regex: Option<&Regex>,
    replacement: Option<&str>,
) -> String {
    let regex = regex.unwrap_or(&regex_constants::ESCAPE_REGEX);
    let replacement = replacement.unwrap_or(" ");

    regex.replace_all(text, replacement).to_string()
}

pub fn replace_numbers(text: &str, replacement: Option<&str>) -> String {
    let replacement = replacement.unwrap_or(" %number% ");
    regex_constants::NUMBERS_REGEX
        .replace_all(text, replacement)
        .to_string()
}

File: text_normalizer/transforms.py
import re
from regex_constants import (
    HTML_REGEX,
    NUMBERS_REGEX,
    ESCAPE_REGEX,
    PUNCTUATION_REGEX,
)
from unidecode import unidecode


def to_lower(text: str) -> str:
    return text.lower()


def remove_accents(text: str) -> str:
    return unidecode(text)


def remove_punctuation(text: str) -> str:
    return re.sub(PUNCTUATION_REGEX, " ", text)


def remove_blank_spaces(text: str) -> str:
    return " ".join(text.split())


def remove_html(text: str) -> str:
    return re.sub(HTML_REGEX, " ", text)


def replace_escape_characters(
    text: str,
    regex: str | None = ESCAPE_REGEX,
    replacement: str | None = " "
) -> str:
    return re.sub(regex, replacement, text)


def replace_numbers(text: str, replacement: str | None = " %number% ") -> str:
    return re.sub(NUMBERS_REGEX, replacement, text)
